src/colmap/sensor/models.h

```diff
@@ -1592,4 +1689,49 @@ double CameraModelCamFromImgThreshold(const CameraModelId model_id,
   return -1;
 }
 
+bool CameraModelIsFisheye(const CameraModelId model_id) {
+  switch (model_id) {
+#define CAMERA_MODEL_CASE(CameraModel) case CameraModel::model_id:
+
+    FISHEYE_CAMERA_MODEL_CASES
+    return true;
+    break;
+    default:
+      return false;
+      break;
+
+#undef CAMERA_MODEL_CASE
+  }
+
+  return false;
+}
+
+bool FisheyeCameraModelIsValidPixel(const CameraModelId model_id,
+                                    const std::vector<double>& params,
+                                    const Eigen::Vector2d& xy) {
+  switch (model_id) {
+#define CAMERA_MODEL_CASE(CameraModel)                 \
+  case CameraModel::model_id: {                        \
+    double uu, vv;                                     \
+    CameraModel::FisheyeCoordinateFromImg(             \
+        params.data(), xy.x(), xy.y(), &uu, &vv);      \
+    const double theta = std::sqrt(uu * uu + vv * vv); \
+    if (theta < M_PI / 2.0)                            \
+      return true;                                     \
+    else                                               \
+      return false;                                    \
+  } break;
```

Comment:
break unnecessary

Reply:
Haha thanks. I was following here: https://github.com/colmap/colmap/blob/main/src/colmap/sensor/models.h#L1585. Will update that as well tonight : )